### PR TITLE
 notifier: adding basic oncall notifications

### PR DESCRIPTION
**Pull request by alyssawilk**
Since slack google calendar support is deprecated, adding it back in via the PR notifier.
(theoretically) notifying general and oncall of rotation switches.  This won't work (but will post that it doesn't work) if we do overrides incorrectly.

### CODE REVIEW

**Comment by alyssawilk** [comment]
@phlax per usual sorry for my python
code is ready for review but PR is not ready to submit - I haven't updated the requirements and I'm not sure how to test that I've added them correctly.  can you clue me in?

**Comment by alyssawilk** [comment]
ah, progress, found the in file, trying to figure out why I can't run pip-compile to fix the txt

**Comment by phlax** [comment]
>  pip-compile to fix the txt

```console
$ pip install pip-tools
$ cd $DIR_CONTAINING_REQS
$ pip-compile --generate-hashes --reuse-hashes --allow-unsafe requirements.in
```

**Comment by alyssawilk** [comment]
yeah for some reason installing pip-compile didn't work.  possibly python versioning, possibly google internal weirdness
python -m piptools compile --generate-hashes worked well enough with some minor differences I think I've cleaned up

**Comment by alyssawilk** [comment]
Eventual plan follow-up: associating opsgenie names with slack names so we can ping ocall from both envoy-maintainers and envoy-CI the week of rotation, and also pinging next-oncall the week before to give them a heads up rotaton will switch to them

File: .github/actions/pr_notifier/pr_notifier.py [hunk 1]
```diff
@@ -226,10 +228,33 @@ def post_to_oncall(client, unassigned_prs, out_slo_prs):
...
+        if datetime.date.today().weekday() == 0:
+            oncall = parse_calendar()
+            client.chat_postMessage(channel='#envoy-maintainer-oncall', text=(oncall))
+            client.chat_postMessage(channel='#general', text=(oncall))
     except SlackApiError as e:
         print("Unexpected error %s", e.response["error"])
 
 
+def parse_calendar():
+    e = requests.get(
```
Comment (phlax) [inline .github/actions/pr_notifier/pr_notifier.py:241]:
```suggestion
    response = requests.get(
```

`e` is usually reserved for errors as in `except Exception as e`

File: .github/actions/pr_notifier/pr_notifier.py [hunk 2]
```diff
@@ -226,10 +228,33 @@ def post_to_oncall(client, unassigned_prs, out_slo_prs):
...
+        "https://calendar.google.com/calendar/ical/d6glc0l5rc3v235q9l2j29dgovh3dn48%40import.calendar.google.com/public/basic.ics"
```
Comment (phlax) [inline .github/actions/pr_notifier/pr_notifier.py:242]: i think lets make this a CONSTANT at top of file

File: .github/actions/pr_notifier/pr_notifier.py [hunk 3]
```diff
@@ -226,10 +228,33 @@ def post_to_oncall(client, unassigned_prs, out_slo_prs):
...
+    )
+    ecal = icalendar.Calendar.from_ical(e.text)
+
+    now = datetime.datetime.now()
+    sunday = now - datetime.timedelta(days=now.weekday() + 1)
+
+    for component in ecal.walk():
+        if component.name == "VEVENT":
+            if (sunday.date() == component.decoded("dtstart").date()):
+                e.close()
```
Comment (phlax) [inline .github/actions/pr_notifier/pr_notifier.py:252]: im not so familiar with the github api - i usually use gidgethub as its async - but i think you can close this connection immediately (wondering if its necessary at all)

**Review by phlax — DISMISSED** [review]
assuming

> NOT YET READY TO SUBMIT

is out of date

lgtm, thanks @alyssawilk

**Review by phlax — APPROVED** [review]

**Pull request merged**